dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/NearestNeighbor.java

```diff
@@ -64,7 +75,30 @@ public static void main(String[] args) {
         System.out.printf("Elapsed time: %d seconds\n", Math.round(elapsed / 1000));
 
         System.out.println(String.format("Closest neighbor was %s", closest.toString()));
+ public static int partition(Point[] arr, int low, int high) {
+        Point pivot = arr[high];
+        int i = low - 1;
+
+        for (int j = low; j < high; j++) {
+            if (calculateDistance(arr[j], target) < calculateDistance(pivot, target)) {
+                i++;
+                Point temp = arr[i];
+                arr[i] = arr[j];
+                arr[j] = temp;
+            }
+        }
+
+        Point temp = arr[i + 1];
+        arr[i + 1] = arr[high];
+        arr[high] = temp;
 
+        return i + 1;
+    }
+} public static double calculateDistance(Point point1, Point point2) {
+    int deltaX = point1.getX() - point2.getX();
+    int deltaY = point1.getY() - point2.getY();
+    return Math.sqrt(deltaX * deltaX + deltaY * deltaY);
+}
```

Comment:
What is being done seems to be that the for-loop checks every point on the list for it's distance from the target and sorts it by comparing the last checked point with the current one. The calculateDistance is what's being called to check that distance.
The variable names of deltaX and deltaY makes this really readable.